### ISSUE TITLE
Installation incompatibility with Miraheze setup

### ISSUE THREAD

**Issue by Routhwick** [SemanticMediaWiki/SemanticResultFormats#732]
### Setup

- MW version: [1.38.2](https://mediawiki.org/wiki/MediaWiki_1.38) (c40e48d)
- DB (MySQL etc.): 10.5.17-MariaDB-1:10.5.17+maria~deb11-log
- PHP version: 7.4.30 (fpm-fcgi)
- SMW version: 4.0.2 (15775c9)
- SRF version: 4.0.1

### Issue

Last month, [Miraheze turned down Semantic Result Formats](https://phabricator.miraheze.org/T9287) due to incompatibility with their submodule-based approach to installing extensions. When @Universal-Omega attempted to install SRF on Miraheze during the mid-afternoon of September 12, the following backtrace resulted. (Emphasis in UO's commentary bolded.)

`
Fatal error: Cannot redeclare smwfContLang() (previously declared in /srv/mediawiki/w/extensions/SemanticResultFormats/extensions/SemanticMediaWiki/includes/GlobalFunctions.php:28) in /srv/mediawiki/w/extensions/SemanticMediaWiki/includes/GlobalFunctions.php on line 27
`

> "I believe this is caused by an incompatibility with how composer is needed to be installed for the extension, **which is incompatible with our setup of submodules**.
>
> "Unfortunately, because of that, that extension can not be done."

A little while later, UO added:

> "I may look into adjusting our composer setup eventually to support those other type of extensions, but for now they can not be installed. I apologise for the inconvenience."

After I brought up [this Hostinger support page](https://support.hostinger.com/en/articles/1583195-how-to-fix-php-fatal-error-cannot-redeclare) to his attention, he elaborated further

> "The submodules way we install extensions mean we can not edit extensions, the way MediaWiki also works, prevents us from modifying the way things load, so in this case, that will not work. The issue is because SMW is installed via submodule, **but the extension tries to re-install and include it via composer, which leads to files being double loaded** and thus not working."

After nearly a month's delay (due to babysitting + wiki-maintenance commitments et al.), I am making this filing on their behalf, and will send them any received feedback/concerns en route to tentative reconsideration. (I am a user of SMW on Miraheze, and also utilised SRF back when Referata had that as a companion extension.)

TL/DR: Any way to retrofit SRF's code so that Miraheze's submodules can handle it?

**Comment by JeroenDeDauw** [comment]
Perhaps they can work around the issue by telling Composer that SMW is already installed. That said, installing the dependencies of MediaWiki and extensions independently with Composer is not safe. Composer is supposed to know about all dependencies and only install a single copy of each.
